### ISSUE TITLE
Re-enable monitor after disabling it with hyprctl

### ISSUE THREAD

**Issue by niraami** [hyprwm/Hyprland#2549]
Before I write a script for this, I thought it would be a good idea to pitch it as a feature - or maybe it already is, but I can't figure it out.
I'd like to be able to re-enable a monitor after I've disabled it using `hyprctl keyword monitor <input>,disable`. The only way do this right now I believe, is to parse the `hyprland.conf`, search for the monitors configuration, and set it using `hyprctl keyword monitor` again - or, if it is not defined in the config, use defaults.

Is this something that could be done? Perhaps by adding `hyprctl keyword mointor <input>,enable`? I guess adding it as a feature would be better than a script, as changes not stored in `hyprland.conf` could be restored when being enabled.

**Comment by MightyPlaza** [comment]
http://wiki.hyprland.org/Configuring/Using-hyprctl/#keyword

**Comment by niraami** [comment]
@MightyPlaza Could you elaborate?

**Comment by MightyPlaza** [comment]
hyprctl keyword monitor DP-3,1920x1080@144,0x0,1

**Comment by niraami** [comment]
I very well know that, those parameters at the end have to come from somewhere though, so you either have to parse the `hyprland.conf`, or, store the monitor configuration before disabling it, and use it to restore it when enabling the monitor again.
Which you can do with a script, and I might, but I'd rather it be a Hyprland feature. I just realized you might not see the value in this if you only have one monitor - then the script is simple, but what if you have three? Or what if you have 1, and then a dock you sometimes plug into, and you want to disable the internal display if you do so. Or what if you change the monitor configuration in the `hyprland.conf` - if you're not using auto/preferred. You now have to change all of your scripts as well.

I thought I outlined this pretty well in my original description, sorry for the confusion.

**Comment by MightyPlaza** [comment]
Maybe put in hyprland.conf
```
env = mon2info,1920x1080@144,0x0,1
```
then enable the monitor with
```
hyprctl keyword monitor DP-3, $mon2info
```

**Comment by niraami** [comment]
I've made a script that does all that I'd expect from such a feature, and probably even a bit too much as I've handled most edge cases as well.

https://gist.github.com/niraami/7f6b3d00cbff485908fea4f631bcedf7

**Comment by skogler** [comment]
Does re-enabling work for you? For me hyprctl does not  list the monitor anymore after I run disable, and the keyword monitor does not turn it on again.

**Comment by niraami** [comment]
Yes it does work for me, you just can't use the ID of the monitor (as you said, it's not listed), but the output (name) instead. Something like `hyprctl monitor DP-2,preferred,auto,1` should work, just replace `DP-2` with your desired output.

But I'm curious, does `hyprctl reload` work, or do you have to do a full restart?  
Also, this might have something to do with DPMS or the monitor itself, as one of my monitors also, very rarely, doesn't turn back on If I spam DPMS commands too fast.

**Comment by skogler** [comment]
Thanks, `hyprctl reload` helps! 

Re-enabling it with `hyprctl keyword monitor DP-3,preferred,auto,1`  works for my main monitor, but not for my second one (the one I need it for, unfortunately).

Main monitor: LG 27GL850
Second one: AOC Q27B3MA

**Comment by k4j8** [comment]
`hyprctl reload` works for me, thanks niraami! I have three monitors that I enable/disable with bash scripts depending on whether I need them or not. Previously, they would not come back after disabled, but putting `hyprctl reload` as the first line in the scripts fixed the issue for me.

**Comment by theotrst** [comment]
Hi,
I just came by. I am a newbie and I am looking for a solution to able/disable monitor. I know it is already an issue and I guess my case is quite similar to many others. In hyprland.conf I have tried something like:

monitor = HDMI-A-1,3440x1440,0x0,1
monitor = eDP-1,1920x1080,760x1441,1
bind = SUPERCTRL,8,exec,hyprctl keyword monitor HDMI-A-1,preferred,auto,1 && keyword monitor eDP-1,disable,1
bind = SUPERCTRL,9,exec,hyprctl keyword monitor HDMI-A-1,disable,1 && keyword monitor eDP-1,preferred,auto,1
bind = SUPERCTRL,0,exec,hyprctl reload

it partially works: I can disable the external one (HDMI) leaving the default monitor (eDP) and I can reload the initial setting. What appparently does not work is the first bind: disable the default monitor (eDP) and leave the external (HDMI).

Any ideas?
many thanks

**Comment by MightyPlaza** [comment]
your command is wrong
you need to use type hyprctl again or use batch

**Comment by theotrst** [comment]
thanks. Done! it works. 
[even if it seems that I cannot switch directly between "single-monitor settings" (HDMI-on/eDP-off ; HDMI-off/eDP-on), I need to reload first the configuration with both monitors]

**Comment by romanstingler** [comment]
@vaxerski can be closed

**Comment by niraami** [comment]
If this is something that doesn't fit the scope of the project, then I'm satisfied with it being closed. I just haven't seen such sentiment yet, so I left it open for more feedback.  
I have solved my use case, and was actually planning on solving it with a script from the get-go, but I thought it might benefit the project.

**Comment by ashish-kus** [comment]
```
#!/bin/bash

INTERNAL_MONITOR="eDP-1"
EXTERNAL_MONITOR=$(hyprctl -j monitors | jq -r ".[] | select(.name != \"$INTERNAL_MONITOR\") | .name" | head -n 1)
MONITOR_CONF="/home/ashishk/.config/hypr/hyprland/monitors.conf"

apply_monitor_config() {
  while IFS= read -r line; do
    [[ "$line" =~ ^[[:space:]]*# ]] && continue
    [[ "$line" =~ ^[[:space:]]*$ ]] && continue
    if [[ "$line" =~ ^monitor[[:space:]]*=[[:space:]]*(.*) ]]; then
      hyprctl keyword monitor "${BASH_REMATCH[1]}"
    fi
  done <"$MONITOR_CONF"
}

acpi_listen | while read -r event; do
  case "$event" in
  *LID\ close*)
    echo "Lid closed"
    hyprctl keyword monitor "$INTERNAL_MONITOR,disable"
    ;;
  *LID\ open*)
    echo "Lid opened"
    apply_monitor_config
    ;;
  esac
done
```


Any suggestion if i am using such script to manage this.
